sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/MetricsAdvisorAdministrationAsyncClient.java

```diff
@@ -731,7 +734,27 @@ private Mono<PagedResponse<DataFeedIngestionStatus>> listDataFeedIngestionStatus
             context)
             .doOnRequest(ignoredValue -> logger.info("Listing ingestion status for data feed"))
             .doOnSuccess(response -> logger.info("Listed ingestion status {}", response))
-            .doOnError(error -> logger.warning("Failed to ingestion status for data feed", error));
+            .doOnError(error -> logger.warning("Failed to ingestion status for data feed", error))
+            .map(res -> new PagedResponseBase<>(
+                res.getRequest(),
+                res.getStatusCode(),
+                res.getHeaders(),
+                toDataFeedIngestionStatus(res.getValue()),
+                res.getContinuationToken(),
+                null));
+    }
+
+    private List<DataFeedIngestionStatus> toDataFeedIngestionStatus(List<com.azure.ai.metricsadvisor.implementation.models.DataFeedIngestionStatus> ingestionStatusList) {
+        return ingestionStatusList
+            .stream()
+            .map(ingestionStatus -> {
+                DataFeedIngestionStatus dataFeedIngestionStatus = new DataFeedIngestionStatus();
+                DataFeedIngestionStatusHelper.setMessage(dataFeedIngestionStatus, ingestionStatus.getMessage());
+                DataFeedIngestionStatusHelper.setIngestionStatusType(dataFeedIngestionStatus, IngestionStatusType.fromString(toStringOrNull(ingestionStatus.getStatus())));
+                DataFeedIngestionStatusHelper.setTimestamp(dataFeedIngestionStatus, ingestionStatus.getTimestamp());
+                return dataFeedIngestionStatus;
```

Comment:
Is this needed? The DataFeedIngestionStatus should be correctly serialized and we don't need accessors to make it public. What's the difference between the DataFeedIngestionStatus  in implementation package and the one in public package?

Reply:
I think the issue is we need the `DataFeedIngestionStatus` in `admin.models` and not `models`. I believe earlier it was manually moved. But to make the swagger work as is without manual modifications we needed this transform.
are there any other options we can do?